kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/MergeQueries.java

```diff
@@ -187,11 +187,11 @@ SELECT ARRAY_AGG(
       )
       ON `<destinationTable>`.<keyField>=src.key
       WHEN MATCHED AND src.value IS NOT NULL
-        THEN UPDATE SET <valueField>=src.value.<field>[, <valueField>=src.value.<field>...]
+        THEN UPDATE SET `<valueField>`=src.value.<field>[, `<valueField>`=src.value.<field>...]
```

Comment:
Thanks @C0urante for your feedback. 

I'm afraid that I'm not aware of any more detailed BigQuery documentation on this apart from the link I already shared. However, let me explain how I tested this:

- With the changes of this branch, I managed to pass all integration tests, being especially relevant `UpsertDeleteBigQuerySinkConnectorIT`.

- I have built a Confluent Hub Archive as per README, and deployed it in a real-world scenario a couple of weeks ago without issues so far.

- Manually in the BigQuery portal. It parses the query nearly in real-time while typing, even without executing it and highlights mistakes. Addressing your comments about when to quote and when don't, I have inferred from this manual testing that the right-hand side doesn't require it - may be because of the dotted syntax (e.g.: `src.value.<field>`) does not produce any ambiguity.

Elaborating a little more in the 3rd, and playing with the SQL example I provided in the first comment, the BQ Query Editor highlights an error for not quoting the word "order":

![error](https://user-images.githubusercontent.com/4973542/150526995-99f595a7-54ad-4a8d-8707-62c2f8e0dfb8.png)

Once fixed, it jumps to the other mistake at line 11. Notice that for BQ line 9 is OK without quoting the right-hand part of the assignment:

![error2](https://user-images.githubusercontent.com/4973542/150527025-6ab3df76-a363-4952-abf6-acb85cbaee78.png)

To sum up, I think that additional quoting will not break anything but might be redundant. In other clauses, as in `GROUP BY`, `ON`, etc. the column identifiers are always part of the dotted syntax and do not appear alone, which I think is the root cause of the issue. 

Thoughts?